File: src/jianzhi40.py
class Solution:
    def getLeastNumbers(self, arr: List[int], k: int) -> List[int]:
        #先用简单的，调库法
        #时间代价就是排序的o(logn)
        #五分钟，一次提交，时间击败：70.90%，空间击败：85.54%
        if len(arr) <= k:
            return arr
        sortedList = sorted(arr)
        return sortedList[:k]
        #还有一种方法是利用快排的切分函数，当被切分的元素刚好是第k个时，意味着左边前k个已经是最小的了
        #最后还有一种是堆排序，那个我用C++做吧

File: src/jianzhi40.cpp
class Solution {
public:
    vector<int> getLeastNumbers(vector<int>& arr, int k) {
        //用优先队列
        //两次通过，十五分钟，时间击败：58.24%，空间击败：33.68%
        if(k == 0){
            vector<int> result;
            return result;
        }
        if(arr.size() <= k){
            return arr;
        }
        else{
            priority_queue<int> pQueue; //优先队列，默认小顶堆
            //首先推入K个
            for(int i = 0; i < k; i++){
                pQueue.push(arr[i]);
            }
            //cout<<2<<endl;
            //然后推入后k个
            for(int i = k; i < arr.size(); i++){
                if(pQueue.top() > arr[i]){
                    //找到更小的元素了
                    pQueue.pop();
                    pQueue.push(arr[i]);
                }
                else{
                    continue;
                }
            }
            //cout<<1<<endl;
            //最后，返回结果
            vector<int> result(k, 0);
            for(int i = 0; i < k; i++){
                result[i] = pQueue.top();
                pQueue.pop();
            }
            return result;
        }
    }
};
